x/babylon/contract/in_message.go

```diff
@@ -1,11 +1,10 @@
 package contract
 
 // CustomMsg is a message sent from a smart contract to the Babylon module
-// TODO: implement
-type CustomMsg struct {
-	Test *TestMsg `json:"test,omitempty"`
-}
-
-type TestMsg struct {
-	Placeholder string `json:"placeholder,omitempty"`
-}
+type (
+	CustomMsg struct {
+		MintRewards *MintRewardsMsg `json:"mint_rewards,omitempty"`
+	}
+	// MintRewardsMsg mints block rewards to the staking contract
+	MintRewardsMsg struct{}
+)
```

Comment:
Since these will be triggered upon `BeginBlock` anyway, wdyt about doing the inflation entirely in `BeginBlock` rather than going through the "module -> contract -> module" workflow?

Reply:
Did it this way because I want rewards to be minted on block finalisation. See https://github.com/babylonlabs-io/babylon-contract/pull/80.

Probably there are simpler ways to detect we have an active FP set that is signing blocks, but this is the strongest one I can see. And yes, it depends on the finality contract.

Reply:
> Did it this way because I want rewards to be minted only on block finalisation. See https://github.com/babylonlabs-io/babylon-contract/pull/80.

This is a product design that I'm not entirely sure. IMO, since every block is expected to be finalised anyway, it's okay to mint tokens upon every block. If we don't want to distribute tokens when no block is finalised, we could instead distribute tokens upon finalising a block. Wdyt?